week08/movie/src/components/Header.js

```diff
@@ -53,17 +56,43 @@ const SignBtn = styled.button`
   font-size: 14px;
   margin-left: 0px;
 `;
-const Header = () => {
+
+const ThemeBtn = styled.button`
+  background: none;
+  border: none;
+  cursor: pointer;
+  color: yellow;
+  width: 30px;
+  height: 30px;
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding-left: 40px;
+  padding-right: 10px;
+  &:before {
+    content: "${(props) => (props.theme.mode === "dark" ? "✨" : "🌙")}";
+    font-size: 20px;
+  }
+`;
+
+const Header = ({ onThemeChange }) => {
+  const theme = useSelector((state) => state.themeSlicer.theme);
+
   return (
-    <HeaderContainer>
-      <HeaderLeft>
-        <Logo>JIN's THEATRE</Logo>
-      </HeaderLeft>
-      <HeaderRight>
-        <HeaderRightText>로그인</HeaderRightText>
-        <SignBtn>회원가입</SignBtn>
-      </HeaderRight>
-    </HeaderContainer>
+    <ThemeProvider theme={theme}>
+      <HeaderContainer>
+        <HeaderLeft>
+          <Logo>JIN's THEATRE</Logo>
+        </HeaderLeft>
+        <HeaderRight>
+          <ThemeBtn onClick={onThemeChange} />
+
+          <HeaderRightText>로그인</HeaderRightText>
+          <SignBtn>회원가입</SignBtn>
+        </HeaderRight>
+      </HeaderContainer>
+    </ThemeProvider>
```

Comment:
상위 컴포넌트인 App에서 이미 `<ThemeProvider theme={theme}>`로 theme 코드를 공급해주었으므로,
하위 컴포넌트인 Header에서 추가로 theme를 공급할 필요가 없습니다.
따라서 `const theme = useSelector((state) => state.themeSlicer.theme);` 이 코드와 `<ThemeProvider>` 코드를 삭제해주시는 것이 좋습니다.

```suggestion
    <HeaderContainer>
      <HeaderLeft>
        <Logo>JIN's THEATRE</Logo>
      </HeaderLeft>
      <HeaderRight>
        <ThemeBtn onClick={onThemeChange} />

        <HeaderRightText>로그인</HeaderRightText>
        <SignBtn>회원가입</SignBtn>
      </HeaderRight>
    </HeaderContainer>
```